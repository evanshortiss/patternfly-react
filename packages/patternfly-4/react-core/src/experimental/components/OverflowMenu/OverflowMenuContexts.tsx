import * as React from 'react';

export const OverflowMenuContext = React.createContext<{
  breakpoint?: 'md' | 'lg' | 'xl'
}>({
  breakpoint: null
});

export const OverflowMenuContentContext = React.createContext<{
  isHidden: boolean
}>({
  isHidden: false
});
