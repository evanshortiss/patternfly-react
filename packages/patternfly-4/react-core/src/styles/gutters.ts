import { getModifier } from '@patternfly/react-styles';

export const GutterSize = {
  sm: 'sm',
  md: 'md',
  lg: 'lg'
};

/**
 * @param styleObj
 * @param size
 * @param defaultValue
 */
export function getGutterModifier(styleObj, size, defaultValue) {
  return getModifier(styleObj, `gutter-${size}`, defaultValue);
}
