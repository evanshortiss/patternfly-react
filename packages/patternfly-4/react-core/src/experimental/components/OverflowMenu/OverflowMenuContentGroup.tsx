import * as React from 'react';
import { css } from '@patternfly/react-styles';
import styles from '@patternfly/react-styles/css/components/OverflowMenu/overflow-menu';
import { OverflowMenuContentContext } from './OverflowMenuConstants';

export interface OverflowMenuContentGroupProps extends React.HTMLProps<HTMLDivElement> {
  /** Any elements that can be rendered in the menu */
  children?: any;
  /** Additional classes added to the OverflowMenuGroup */
  className?: string;
  /** Modifies the overflow menu group visibility */
  persistent?: boolean;
  /** Indicates a button or icon group */
  groupType?: string;
}

export const OverflowMenuContentGroup: React.SFC<OverflowMenuContentGroupProps> = ({
  className,
  children,
  persistent = false,
  groupType = '',
  ...props
}) => (
  <OverflowMenuContentContext.Consumer>
    {value => (
      <div hidden={!persistent && value.isHidden} className={css(
        styles.overflowMenuGroup,
        groupType === 'button' && styles.modifiers.buttonGroup,
        groupType === 'icon' && styles.modifiers.iconButtonGroup,
        persistent && styles.modifiers.persistent,
        className
      )}>
        { children }
      </div>
    )}
  </OverflowMenuContentContext.Consumer>
);