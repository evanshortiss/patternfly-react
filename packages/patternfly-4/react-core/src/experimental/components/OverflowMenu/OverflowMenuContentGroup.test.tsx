import * as React from 'react';
import { mount } from 'enzyme';
import styles from '@patternfly/react-styles/css/components/OverflowMenu/overflow-menu';
import { OverflowMenuContentGroup } from './OverflowMenuContentGroup';
import { OverflowMenuContentContext } from './OverflowMenuContexts';

describe('OverflowMenuContentGroup', () => {
  test('isPersistent and Hidden should still show', () => {
    const view = mount(
      <OverflowMenuContentContext.Provider value={{ isHidden: true }}>
        <OverflowMenuContentGroup isPersistent />
      </OverflowMenuContentContext.Provider>
    );
    expect(view.find(`.${styles.overflowMenuGroup}`).length).toBe(1);
    expect(view.find(`.${styles.modifiers.persistent}`).length).toBe(1);
    expect(view.find({ hidden: true }).length).toBe(0);
    expect(view).toMatchSnapshot();
  });

  test('Hidden but not isPersistent should be hidden', () => {
    const view = mount(
      <OverflowMenuContentContext.Provider value={{ isHidden: true }}>
        <OverflowMenuContentGroup />
      </OverflowMenuContentContext.Provider>
    );
    expect(view.find(`.${styles.modifiers.persistent}`).length).toBe(0);
    expect(view.find({ hidden: true }).length).toBe(1);
    expect(view).toMatchSnapshot();
  });

  test('Button group', () => {
    const view = mount(
      <OverflowMenuContentContext.Provider value={{ isHidden: false }}>
        <OverflowMenuContentGroup groupType="button" />
      </OverflowMenuContentContext.Provider>
    );
    expect(view.find(`.${styles.modifiers.buttonGroup}`).length).toBe(1);
    expect(view).toMatchSnapshot();
  });

  test('Icon group', () => {
    const view = mount(
      <OverflowMenuContentContext.Provider value={{ isHidden: false }}>
        <OverflowMenuContentGroup groupType="icon" />
      </OverflowMenuContentContext.Provider>
    );
    expect(view.find(`.${styles.modifiers.iconButtonGroup}`).length).toBe(1);
    expect(view).toMatchSnapshot();
  });
});
