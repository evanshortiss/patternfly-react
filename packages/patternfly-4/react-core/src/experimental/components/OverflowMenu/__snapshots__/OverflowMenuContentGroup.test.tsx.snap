// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverflowMenuContentGroup Button group 1`] = `
<Component
  groupType="button"
>
  <div
    className="pf-c-overflow-menu__group pf-m-button-group"
    hidden={false}
  />
</Component>
`;

exports[`OverflowMenuContentGroup Hidden but not isPersistent should be hidden 1`] = `
<Component>
  <div
    className="pf-c-overflow-menu__group"
    hidden={true}
  />
</Component>
`;

exports[`OverflowMenuContentGroup Icon group 1`] = `
<Component
  groupType="icon"
>
  <div
    className="pf-c-overflow-menu__group pf-m-icon-button-group"
    hidden={false}
  />
</Component>
`;

exports[`OverflowMenuContentGroup isPersistent and Hidden should still show 1`] = `
<Component
  isPersistent={true}
>
  <div
    className="pf-c-overflow-menu__group pf-m-persistent"
    hidden={false}
  />
</Component>
`;
